src/input/scroll.c

```diff
@@ -243,6 +251,81 @@ touch_coords scroll_get_original_touch_position(void)
     return data.start_touch;
 }
 
+mouse_coords scroll_get_original_mouse_position(void)
+{
+    return data.start_mouse;
+}
+
+void scroll_start_mouse_drag(const pixel_offset *position, mouse_coords coords)
+{
+    data.start_mouse = coords;
+    data.has_scrolled = 0;
+    data.is_scrolling = 1;
+    data.is_touch = 0;
+    data.cumulative_delta_x = 0;
+    data.cumulative_delta_y = 0;
+    clear_scroll_decay(position);
+
+    platform_mouse_set_relative_mode(1);
+
+    // Discard the first value, which is incorrect (the first one gives the relative position to center of window)
+    platform_mouse_get_relative_state(NULL, NULL);
```

Comment:
Move this correction to `platform_mouse_set_relative_mode(1)` so that platform (SDL) implementation details do not leak into the main codebase.
Also don't forget to remove the <stddef.h> include.